### PR TITLE
Refactored HA from vovo4ka

### PR DESCRIPTION
**Pull request by mihgen**

### CODE REVIEW

File: lib/astute/deployment_engine.rb [hunk 1]
```diff
@@ -74,57 +78,105 @@ def attrs_multinode(nodes, attrs)
...
     def deploy_multinode(nodes, attrs)
-      deploy_ha_full(nodes, attrs)
+      ctrl_nodes = nodes.select {|n| n['role'] == 'controller'}
+      compute_nodes = nodes.select {|n| n['role'] == 'compute'}
+
+      Astute.logger.info "Starting deployment of primary controller"
+      deploy_piece(ctrl_nodes, attrs)
+
+      Astute.logger.info "Starting deployment of other nodes"
+      deploy_piece(compute_nodes, attrs)
```
Comment (mihgen) [inline lib/astute/deployment_engine.rb]: I believe we are missing cinder nodes here.

File: lib/astute/deployment_engine.rb [hunk 2]
```diff
@@ -74,57 +78,105 @@ def attrs_multinode(nodes, attrs)
...
           'name'                 => n['fqdn'].split(/\./)[0],
-          'role'                 => 'controller',
-          'internal_address'     => n['network_data'].select {|nd| nd['name'] == 'management'}[0]['ip'].split(/\//)[0],
-          'public_address'       => n['network_data'].select {|nd| nd['name'] == 'public'}[0]['ip'].split(/\//)[0],
+          'role'                 => n['role'],
           'mountpoints'          => "#{attrs['mp']['point']} #{attrs['mp']['weight']}",
-          'zone'                 => n['id'],
-          'storage_local_net_ip' => n['network_data'].select {|nd| nd['name'] == 'storage'}[0]['ip'].split(/\//)[0],
+          'internal_address'     => n['network_data'].select {|nd| select_ifaces(nd['name'], 'management')}[0]['ip'].split(/\//)[0],
+          'internal_br'          => n['internal_br'],
```
Comment (mihgen) [inline lib/astute/deployment_engine.rb:95]: do we pass such param from nailgun?

File: lib/astute/deployment_engine/nailyfact.rb [hunk 3]
```diff
@@ -49,14 +49,24 @@ def create_facts(node, attrs)
...
         iface['dev']
       end
-      metadata["#{iface['name']}_interface"] = device
-      if iface['ip']
-        metadata["#{iface['name']}_address"] = iface['ip'].split('/')[0]
+
+      if iface['name'].is_a?(String)
+        metadata["#{iface['name']}_interface"] = device
+        if iface['ip']
+#         metadata["#{iface['name']}_address"] = iface['ip'].split('/')[0]
```
Comment (mihgen) [inline lib/astute/deployment_engine/nailyfact.rb]: ?

File: lib/astute/deployment_engine.rb (hunk 2 above)
Comment (yoojinl) [inline lib/astute/deployment_engine.rb:95]: No.

Comment (mihgen) [inline lib/astute/deployment_engine.rb:95]: Then we need to check that puppet has defaults for all such params.

Comment (yoojinl) [inline lib/astute/deployment_engine.rb:95]: Same for `public_br`

File: lib/astute/deployment_engine.rb [hunk 4]
```diff
@@ -74,57 +78,105 @@ def attrs_multinode(nodes, attrs)
...
+          'internal_netmask'     => n['network_data'].select {|nd| select_ifaces(nd['name'], 'management')}[0]['netmask'],
+          'public_address'       => n['network_data'].select {|nd| select_ifaces(nd['name'], 'public')}[0]['ip'].split(/\//)[0],
+          'public_br'            => n['public_br'],
+          'public_netmask'       => n['network_data'].select {|nd| select_ifaces(nd['name'], 'public')}[0]['netmask'],
+          'swift_zone'           => n['id'],
+          'storage_address'      => n['network_data'].select {|nd| select_ifaces(nd['name'], 'storage')}[0]['ip'].split(/\//)[0],
+          'storage_netmask'      => n['network_data'].select {|nd| select_ifaces(nd['name'], 'storage')}[0]['ip'].split(/\//)[0],
```
Comment (yoojinl) [inline lib/astute/deployment_engine.rb]: I think `storage_netmask` should not be same as `storage_address`

File: lib/astute/deployment_engine.rb [hunk 5]
```diff
@@ -39,34 +39,38 @@ def method_missing(method, *args)
...
+      attrs['nodes'] = nodes.map do |n|
+        {
+          'fqdn'                 => n['fqdn'],
+          'name'                 => n['fqdn'].split(/\./)[0],
+          'role'                 => n['role'],
+          'internal_address'     => n['network_data'].select {|nd| select_ifaces(nd['name'], 'management')}[0]['ip'].split(/\//)[0],
+          'internal_br'          => n['internal_br'],
+          'internal_netmask'     => n['network_data'].select {|nd| select_ifaces(nd['name'], 'management')}[0]['netmask'],
+          'public_address'       => n['network_data'].select {|nd| select_ifaces(nd['name'], 'public')}[0]['ip'].split(/\//)[0],
+          'public_br'            => n['public_br'],
```
Comment (yoojinl) [inline lib/astute/deployment_engine.rb:54]: `public_br` don't pass from nailgun.

File: lib/astute/deployment_engine.rb (hunk 1 above)
Comment (yoojinl) [inline lib/astute/deployment_engine.rb]: Yep.

File: lib/astute/deployment_engine/nailyfact.rb (hunk 3 above)
Comment (aglarendil) [inline lib/astute/deployment_engine/nailyfact.rb]: we calculate {iface}_address in site.pp from nodes hash. no need to duplicate it here

File: lib/astute/deployment_engine.rb (hunk 5 above)
Comment (aglarendil) [inline lib/astute/deployment_engine.rb:54]: that's ok. It is used only in quantum scenario.

File: lib/astute/deployment_engine.rb (hunk 4 above)
Comment (aglarendil) [inline lib/astute/deployment_engine.rb]: Absolutely.

**Pull request closed without merging**